DuckDuckGo/Freemium/PIR/FreemiumPIRScanResultPolling.swift

```diff
@@ -116,13 +111,31 @@ private extension DefaultFreemiumPIRScanResultPolling {
         }
     }
 
+    /// Starts the polling process for DBP scan results.
+    ///
+    /// It first checks if any results are available using `notifyMatchesFoundOrNoResultsIfConditionsMet()`.
+    /// If no results are found, it starts a repeating timer to poll for results at regular intervals
+    /// using `startRepeatingConditionCheck()`.
```

Comment:
```suggestion
    /// It first checks if any results are available using `checkResultsAndNotifyIfApplicable()`.
    /// If no results are found, it starts a repeating timer to poll for results at regular intervals
    /// using `startPollingTimer()`.
```

But perhaps it's also fine to not use function calls in this comment because the function body is small and simple enough.